### DISCUSSION TITLE
Pyswmm controls does not change SWMM control rules

### DISCUSSION THREAD

**Discussion by Dinho4** [pyswmm/pyswmm#366] [Q&A] [answered]
The following code does not change SWMM control rules; the output file that I am able to access after simulation shows no changes whatsoever, it's it runs simulation without considering my Pyswmm control rules:

```python
from pyswmm import Simulation, Links, Nodes, SystemStats
from datetime import datetime
with Simulation('C:/Users/Malesela/Downloads/CIV5000WProject.inp') as sim:

    Or21_Uctdam = Links(sim)["21"]
    Pond_Uctdam = Nodes(sim)["12"]
    syst_stats = SystemStats(sim)

    sim.step_advance(300)

    for step in sim:
        RRF_Uctdam = syst_stats.runoff_stats["rainfall"]
        Vol_ro_Uctdam = 0.12*RRF_Uctdam*330
        Vol_ex_Uctdam = 31054 - Pond_Uctdam.volume
        if Pond_Uctdam.depth >= 6 and Pond_Uctdam.depth <= 12.5:
            Or21_Uctdam.target_setting = 0.085
        elif Vol_ro_Uctdam > Vol_ex_Uctdam:
            Or21_Uctdam.target_setting = 1.0
```

**Comment by bemcdonnell** [comment]
@Dinho4 are there any control rules in the `[CONTROLS]` section of the model? Can you zip and send the model?  Also is it possible the hydraulic network didn't create the conditions to make the control change?  Any hydrographs you could add?

**Reply by Dinho4** [reply]
There are control rules in the SWMM Model Section. I have send the model.

**Comment by Dinho4** [comment]
-

**Reply by bemcdonnell** [reply]
@Dinho4 just so you are aware, You can't attached thing to an email in GitHub. You have to zip the file up and log into the comments here on the site to deliver files.  

On the subject of control - if you have a `RULE` in your `[CONTROLS]` section, that is rather than PySWMM's control.  So I suggest you try to migrate all of the control logic pertaining to that link over to your PySWMM control loop.

**Reply by Dinho4** [reply]
And how do I migrate all of the control logic to my PySWMM control loop?
[CIV5000WProject.zip](https://github.com/OpenWaterAnalytics/pyswmm/files/9657960/CIV5000WProject.zip)

**Reply by Dinho4** [reply]
With respect to hydrographs, my control rules in PySWMM should be able to change the Flooding losses for the following node series;

with Output('C:/Users/Malesela/Downloads/CIV5000WProject.out') as out:
...     ts = out.node_series('12', NodeAttribute.FLOODING_LOSSES, datetime(2016, 7, 20, 0, 9, 59), datetime(2016, 7, 21, 23, 54, 59))
...     for index in ts:
...         print(index, ts[index])
...
2016-07-20 00:09:59 0.0
2016-07-20 00:14:59 0.0
2016-07-20 00:19:59 0.0
2016-07-20 00:24:59 0.0
2016-07-20 00:29:59 0.0
2016-07-20 00:34:59 0.0
2016-07-20 00:39:59 0.0
2016-07-20 00:44:59 0.0
2016-07-20 00:49:59 0.0
2016-07-20 00:54:59 0.0
2016-07-20 00:59:59 0.0
2016-07-20 01:04:59 0.0
2016-07-20 01:09:59 0.0
2016-07-20 01:14:59 0.0
2016-07-20 01:19:59 0.0
2016-07-20 01:24:59 0.0
2016-07-20 01:29:59 0.0
2016-07-20 01:34:59 0.0
2016-07-20 01:39:59 0.0
2016-07-20 01:44:59 0.0
2016-07-20 01:49:59 0.0
2016-07-20 01:54:59 0.0
2016-07-20 01:59:59 0.0
2016-07-20 02:04:59 0.0
2016-07-20 02:09:59 0.0
2016-07-20 02:14:59 0.0
2016-07-20 02:19:59 0.0
2016-07-20 02:24:59 0.0
2016-07-20 02:29:59 0.0
2016-07-20 02:34:59 0.0
2016-07-20 02:39:59 0.0
...

**Reply by bemcdonnell** [reply]
@Dinho4 let me have a look at your INP.  Meanwhile, i think it's best if you can plot the data rather than drop the timeseries here

**Comment by bemcdonnell** [comment, marked as answer]
```
[CONTROLS]
RULE 1
IF SIMULATION CLOCKTIME > 19:00:00
AND NODE 12 DEPTH >= 6
THEN ORIFICE 21 SETTING = 0.085
ELSE ORIFICE 21 SETTING = 0
```

SWMM is executing this control logic at 19:00h after your simulation starts.  Therefore, if this is the only logic you have in your model.. .and you're only trying to control Link `21`, just delete this control logic from your model and the pyswmm controls will execute perfectly.

**Reply by bemcdonnell** [reply]
@Dinho4, let me know if this works. If so, I will close the discussion

**Reply by Dinho4** [reply]
Yes, it works. Thanks @bemcdonnell
